### DISCUSSION TITLE
Help Getting Started - Pentair and AquaRite data...

### DISCUSSION THREAD

**Discussion by kjf77** [tagyoureit/nodejs-poolController#903] [App and O/S Config Questions]
Just getting into this in an effort to access my Hayward AquaRite as the rest of the pool already has a Pentair EasyTouch 4 and I can access that data and get it into Home Assistant.  But still confused as I read over the docs for this.
Since my Pentair turns the AquaRite on and off should the AquaRite data be already accessible from the Pentair data?  I also read that version 8 of the pool controller says RS485 is no longer required but then adds nothing to detail how to use it without 485 How does that work?  I also already have the Pentair data being brought indoors to their wireless box with the RS485 port and I could connect a Pi to that but really need the AquaRite data and would like to get to that indoors without mounting a Pi near the pad.  Is any of that doable?

**Comment by tagyoureit** [comment]
Yes, all doable.  njsPC v8 _can_ read data directly from Screenlogic, but it works best with RS485 and you have full control over your system.  You can connect to the RS485 on your Screenlogic wireless unit.  You will also be able to set the set points for your Aquarite with njsPC because it is compatible with Pentair equipment.

**Comment by kjf77** [comment]
Thanks for the reply. Glad to hear it’s doable!So… how does the AquaRite data get from the Hayward to the Pentair?Is that something I need to connect? I haven’t really looked to the board itself. All I know is when I turn off the Pentair the AquaRite turns off too. I already have a rs485 to usb adapter so can I use that to connect the ScreenLogic unit to a Pi then run njsPC on that Pi?Finally is there a way to run njsPC on a Pi that’s  running Home Assistant OS or do I need one that’s running full Pi OS?KenOn Feb 21, 2024, at 8:41 PM, tagyoureit ***@***.***> wrote:﻿
Yes, all doable.  njsPC v8 can read data directly from Screenlogic, but it works best with RS485 and you have full control over your system.  You can connect to the RS485 on your Screenlogic wireless unit.  You will also be able to set the set points for your Aquarite with njsPC because it is compatible with Pentair equipment.

**Comment by tagyoureit** [comment]
You can run njsPC in Home Assistant: https://github.com/Crewski/njsPC-HA

Aquarite is already talking to the Pentair system.  In this case, we will query the EasyTouch for the chlorinator information.  We won't control it directly because then the chlorinator would be trying to take instructions from 2 different control systems.

**Comment by kjf77** [comment]
Hello again,

So I opened the Hayward Aquarite and I see there is nothing attached to the 485 connector.

That would seem to mean that the salt cell isn’t currently connected to the Pentair.

 

I think they are just controlling the power to it as the main AC line seems to be coming from the Pentair panel.

 

And if that is true can I just run my own wire from the Hayward 485 bus to the Pentair?
Not sure where I’d connect in there but I’m sure you can tell me that.

 

Also I had wanted to see just what’s already on the ScreenLogic connecting to that 485 port.

I have a 485 to USB connector but only a Windows laptop to start with.

I’ll eventually connect it to the Home Assistant OS Pi.

Or maybe I’ll just try to get the nodejspool HA version running but guessing there may be issues firing out USB/tty ports unless your software figures that out on the fly.

 

But the Wiki for nodejs-pool mentions Windows in the QuickStart seems to have no detail on that.

Is there a list of what and how to get it running on Windows?

 

Thanks again,

Ken

**Reply by tagyoureit** [reply]
The instructions are the same for Windows/RPi.  https://github.com/tagyoureit/nodejs-poolController?tab=readme-ov-file#prerequisites

The Hayward [manual](https://hayward.com/media/akeneo_connector/asset_files/0/9/092623_RevD_a5aa.pdf) has the wiring for connecting the cell to a Pentair system.

**Comment by kjf77** [comment]
I’m getting nowhere at all and just so you know I have 30+ years experience in software.

 

You said I can just install njsPC-HA directly into HA using HACS.

Doesn’t seem to work at all.  I did the custom repository thing with the github URL and nothing happens.

I get to a page describing it but he says it integrates to nodejs-PoolCOntroller so that sounds like I still need that somewhere.

 

In case I’m not clear this is what I have:

 

Hayward Accurite not currently connected to Pentair but I plan to.

Pentair ScreenLogic box in the house.

RPi running Home Assistant OS sitting next to ScreenLogic.

RS485 from ScreenLogic cabled to RPi USB port.

 

Home Assistant has Pentair integration already bringing in that data but that’s the remote rout and doesnot have Accurite data capability to best of my knowledge.

 

So still not convinced this will work as-is.

 

1.	Can’t get njsPC-HA to fully install to even begin to configure it.
2.	Is the full nodejs-PoolController needed standalone anywhere?
3.	If so where?  Can I get it on that Pi?

 

Hope you can shed some light to end my frustration and confusion.

 

Thanks,

Ken

**Comment by rstrouse** [comment]
I think you may have missed a primary point.  nodejs-PoolController must be installed somewhere.   It is a server platform that communicates with your EasyTouch to translate the raw RS485 data into a usable data and control source.  It acts as a remote on the RS485 bus to the EasyTouch.  Once that is installed and configured, the Home Assistant integration will find njsPC on your network and reflect the settings for your pool in Home Assistant to allow you to control everything and reflect its state.

In short you need njsPC installed and while you are at it I would also install dashPanel as this is a robust interface that will allow you to change the configuration and manage the pool.  Once njsPC is talking to EasyTouch then the chlorinator will automatically show up in Home Assistant.  Both njsPC and dashPanel can be installed on a Windows, Linux (raspberry pi... etc), or Mac.

**Comment by kjf77** [comment]
Ok thanks so much for clarifying.At first I misunderstood but then realized that too.So if you don’t mind me asking a few more questions…In order for the Acurite data to be added I'll have to connect from the rs485 connector on the Accurite to the Intellitouch correct?If so I can’t find clear Pentair instructions on where. Any info on that?The intellitouch has a ScreenLogic box in the house next to my router and my Pi running Home Assistant OS.Can I somehow get njsPC running on that Pi or is that impossible that way?Wondering if i need to instead run njsPC and HA on the Pi but change the OS back to regular Pi OS.KenOn Feb 24, 2024, at 6:57 PM, rstrouse ***@***.***> wrote:﻿
I think you may have missed a primary point.  nodejs-PoolController must be installed somewhere.   It is a server platform that communicates with your EasyTouch to translate the raw RS485 data into a usable data and control source.  It acts as a remote on the RS485 bus to the EasyTouch.  Once that is installed and configured, the Home Assistant integration will find njsPC on your network and reflect the settings for your pool in Home Assistant to allow you to control everything and reflect its state.
In short you need njsPC installed and while you are at it I would also install dashPanel as this is a robust interface that will allow you to change the configuration and manage the pool.  Once njsPC is talking to EasyTouch then the chlorinator will automatically show up in Home Assistant.  Both njsPC and dashPanel can be installed on a Windows, Linux (raspberry pi... etc), or Mac.

**Reply by rstrouse** [reply]
I assume Acurite was a typo.  If you mean AquaRite then IntelliTouch should be compatible with it already.  You simply need to enable IntelliChlor on the IntelliTouch.  The protocol is identical and it should work just fine if you have the RS485 connected to one of the ports on the IntelliTouch.  Then if you connect njsPC also to the RS485 bus then it will pick it up without issue.

**Comment by kjf77** [comment]
Yes sorry that was a typo. But my biggest issue is can I install njsPC on the Pi running Home Assistant OS and if so how, or do I need to either change that pi to raspberry pi OS, or install it on a completely separate Pi?KenOn Feb 24, 2024, at 10:20 PM, rstrouse ***@***.***> wrote:﻿
I assume Acurite was a typo.  If you mean AquaRite then IntelliTouch should be compatible with it already.  You simply need to enable IntelliChlor on the IntelliTouch.  The protocol is identical and it should work just fine if you have the RS485 connected to one of the ports on the IntelliTouch.  Then if you connect njsPC also to the RS485 bus then it will pick it up without issue.

**Comment by kjf77** [comment]
One more thing to mention.If it matters, I actually have an Easytouch not Intellitouch.Will that connect to the AquaRite?Easytouch wiring manual mentions nothing other than relay control to an Intellichlor, nothing about data.Here’s what’s wired in it now.  Looks to me that J20 is the only com port and it already has 2 things connected.  Wondering if I can just add the AquaRite to the same connector.KenOn Feb 24, 2024, at 10:20 PM, rstrouse ***@***.***> wrote:﻿
I assume Acurite was a typo.  If you mean AquaRite then IntelliTouch should be compatible with it already.  You simply need to enable IntelliChlor on the IntelliTouch.  The protocol is identical and it should work just fine if you have the RS485 connected to one of the ports on the IntelliTouch.  Then if you connect njsPC also to the RS485 bus then it will pick it up without issue.

**Reply by rstrouse** [reply]
I know there is a way to monkey around with HAOS using the supervisor but that requires somebody with waaaay more docker experience than me and more fiddling than I want to undertake.   If I were you I would stand up a pi next to HAOS for such things or use a mini-pc with Proxmox for installing VMs.  njsPC will run on anything that can support nodeJS.  If you have a PC you can even install it there.

IntelliTouch and EasyTouch are almost protocol identical.  But when it comes to the chlorinator they are identical so it does not matter whether it is EasyTouch or IntelliTouch.   That is unless you are unlucky enough to have one of the Lite versions of EasyTouch.  That controller is actually closer to SunTouch which is very hobbled.

**Comment by kjf77** [comment]
But can I connect to that same j20 com port?On Feb 25, 2024, at 12:03 PM, rstrouse ***@***.***> wrote:﻿
I know there is a way to monkey around with HAOS using the supervisor but that requires somebody with waaaay more docker experience than me and more fiddling than I want to undertake.   If I were you I would stand up a pi next to HAOS for such things or use a mini-pc with Proxmox for installing VMs.  njsPC will run on anything that can support nodeJS.  If you have a PC you can even install it there.
IntelliTouch and EasyTouch are almost protocol identical.  But when it comes to the chlorinator they are identical so it does not matter whether it is EasyTouch or IntelliTouch.   That is unless you are unlucky enough to have one of the Lite versions of EasyTouch.  That controller is actually closer to SunTouch which is very hobbled.

**Reply by rstrouse** [reply]
Yes RS485 is a bus topology.  So long as the connections are solid you can connect many devices on the same port.

**Comment by kjf77** [comment]
Connected the 2 per the diagrams I’ve seen online and I’m getting the dreaded remote led on for about 4 seconds then offer about 15-20. It shows the salt level in the pentair app but doesn’t seem to allow any control. I tried to turn on super chlorinate and it ignored that. It also seems to only show 50% on the dial. Slightly disappointing. On Feb 25, 2024, at 3:27 PM, rstrouse ***@***.***> wrote:﻿
Yes RS485 is a bus topology.  So long as the connections are solid you can connect many devices on the same port.
